include/lsst/afw/detection/Psf.h

```diff
@@ -283,13 +344,11 @@ class Psf : public daf::base::Citizen,
     virtual geom::Box2I doComputeBBox(geom::Point2D const& position, image::Color const& color) const = 0;
     //@}
 
+
     bool const _isFixed;
-    mutable std::shared_ptr<Image> _cachedImage;
-    mutable std::shared_ptr<Image> _cachedKernelImage;
-    mutable image::Color _cachedImageColor;
-    mutable image::Color _cachedKernelImageColor;
-    mutable geom::Point2D _cachedImagePosition;
-    mutable geom::Point2D _cachedKernelImagePosition;
+    typedef utils::Cache<detail::PsfCacheKey, std::shared_ptr<Image>> PsfCache;
+    mutable std::shared_ptr<PsfCache> _imageCache;
+    mutable std::shared_ptr<PsfCache> _kernelImageCache;
```

Comment:
I believe these no longer need to be `mutable`; the extra layer of indirection does that for you.  Since `Psf` is already noncopyable and immutable, `std::unique_ptr` should be used instead of `std::shared_ptr` as well.